packages/wallets/solflare/src/metamask/wallet.ts

```diff
@@ -0,0 +1,149 @@
+import { WalletNotConnectedError } from '@solana/wallet-adapter-base';
+import { SOLANA_CHAINS } from '@solana/wallet-standard-chains';
+import {
+    SolanaSignAndSendTransaction,
+    type SolanaSignAndSendTransactionFeature,
+    type SolanaSignAndSendTransactionMethod,
+    SolanaSignMessage,
+    type SolanaSignMessageFeature,
+    type SolanaSignMessageMethod,
+    SolanaSignTransaction,
+    type SolanaSignTransactionFeature,
+    type SolanaSignTransactionMethod,
+} from '@solana/wallet-standard-features';
+import type { default as SolflareMetaMask } from '@solflare-wallet/metamask-sdk';
+import type { Wallet } from '@wallet-standard/base';
+import {
+    StandardConnect,
+    type StandardConnectFeature,
+    type StandardConnectMethod,
+    StandardDisconnect,
+    type StandardDisconnectFeature,
+    type StandardDisconnectMethod,
+    StandardEvents,
+    type StandardEventsChangeProperties,
+    type StandardEventsFeature,
+    type StandardEventsListeners,
+    type StandardEventsNames,
+    type StandardEventsOnMethod,
+} from '@wallet-standard/features';
+import { icon } from './icon.js';
+
+export class SolflareMetaMaskWallet implements Wallet {
+    readonly #listeners: { [E in StandardEventsNames]?: StandardEventsListeners[E][] } = {};
+    readonly #version = '1.0.0' as const;
+    readonly #name = 'MetaMask' as const;
+    readonly #icon = icon;
+    #solflareMetaMask: SolflareMetaMask | null = null;
+
+    get version() {
+        return this.#version;
+    }
+
+    get name() {
+        return this.#name;
+    }
+
+    get icon() {
+        return this.#icon;
+    }
+
+    get chains() {
+        return SOLANA_CHAINS.slice();
```

Comment:
Can you confirm which chains the Snap actually supports so we can limit this?

Reply:
The Snap itself is just a signing interface, just like a Ledger device, but the supporting infrastructure (Widget, web wallet etc) should allow switching the network from the settings.

Reply:
Sure, but I'm asking which networks are actually implemented, today, in that web wallet. Dapps need to be able to know where and how to send transactions.